velox/exec/tests/AggregationTest.cpp

```diff
@@ -1351,5 +1351,59 @@ TEST_F(AggregationTest, outputBatchSizeCheckWithSpill) {
   }
 }
 
+TEST_F(AggregationTest, distinctWithSpilling) {
+  auto vectors = makeVectors(rowType_, 10, 100);
+  createDuckDbTable(vectors);
+  auto tempDirectory = exec::test::TempDirectoryPath::create();
+  auto task = AssertQueryBuilder(duckDbQueryRunner_)
+                  .config(QueryConfig::kSpillEnabled, "true")
+                  .config(QueryConfig::kAggregationSpillEnabled, "true")
+                  .config(QueryConfig::kSpillPath, tempDirectory->path)
+                  .config(QueryConfig::kTestingSpillPct, "100")
+                  .plan(PlanBuilder()
+                            .values(vectors)
+                            .singleAggregation({"c0"}, {}, {})
+                            .planNode())
+                  .assertResults("SELECT distinct c0 FROM tmp");
+  auto stats = task->taskStats().pipelineStats;
+  // Verify that spilling is not triggered.
+  ASSERT_EQ(stats[0].operatorStats[1].spilledBytes, 0);
+}
+
+TEST_F(AggregationTest, preGroupedAggregationWithSpilling) {
+  auto vectors = makeVectors(rowType_, 10, 4);
```

Comment:
It looks like the generated vectors are not used as the loop below re-generates all the data. Consider using makeRowVector + makeFlatVector methods instead.

```
makeRowVector({
   // Pre-grouped key.
   makeFlatVector(1'000, [&] (auto /*row*/) { return val++ / 5; }),
   // Payload.
   makeFlatVector(1'000, [] (auto row) {return row}),
});
```